metadata_mapper/mappers/oac_mapper.py

```diff
@@ -9,35 +9,22 @@
 class OacRecord(Record):
 
     def to_UCLDC(self):
-        mapped_data = {"sourceResource": {}}
+        mapped_data = {}
 
-        id = self.source_metadata.get("id", "")
+        id = self.pre_mapped_data.get("id", "")
         mapped_data.update({
             "id": id,
-            "_id": self.source_metadata.get("_id"),
-            "@id": f"http://ucldc.cdlib.org/api/items/{id}",
-            "originalRecord": self.source_metadata.get("originalRecord"),
-            "ingestDate": self.source_metadata.get("ingestDate"), 
-            "ingestType": self.source_metadata.get("ingestType"), 
-            "ingestionSequence": self.source_metadata.get("ingestionSequence"),
+            "originalRecord": self.source_metadata,
             # This is set in select-oac-id but must be added to mapped data
-            'isShownAt': self.source_metadata.get('isShownAt', None),
-            'provider': self.source_metadata.get('provider'),
-            'dataProvider': self.source_metadata.get(
-                'originalRecord', {}).get('collection'),
+            'isShownAt': self.pre_mapped_data.get('isShownAt', None),
             'isShownBy': self.get_best_oac_image(),
-            'item_count': self.map_item_count(),
         })
 
-        mapped_data = self.remove_if_none([
-            "provider",
-            "dataProvider",
-            "isShownBy",
-            "item_count",
-        ], mapped_data)
+        if mapped_data["isShownBy"] is None:
+            mapped_data.pop("isShownBy")
```

Comment:
Do we need to do the same for if isShownAt is None?

Reply:
OAC mapper kind of needs to be completely revisited. At some point. I don't think this makes sense to look at for now though.